documentation/restler/MakeTestProxyRecording.md

```diff
@@ -0,0 +1,167 @@
+
+The general instructions for running RESTler can be found [here](./QuickStart.md).
+There are some specific instructions for running RESTler with test-proxy.
+
+### Update engineSettings.json
+
+You need to modify the `engineSettings.json` file to use the test-proxy.
+
+```sh
+jq '.host = "localhost" | .target_port = 5000 | .no_ssl = true' restlerConfig/engine_settings.json > engine_settings.json
+```
+
+### Update the Restler custom dictionary with test-proxy headers
+
+The test-proxy requires custom headers to be sent with each request that tell it how to handle the request.
+One of these headers specifies the "recording id", which is obtained by sending a request to the test-proxy to start a new recording session.
+
+To start a recording session, send a `/Record/Start` request to the test-proxy.
+The request should specify the name of the recording file and the path to the `assets.json` file.
+The recording file will be created in the `.assets` folder in the root of the azure-rest-api-specs repo,
+but must have a unique name to avoid conflicts with other recordings.
+Here we make the name unique by prepending the same path as the `assets.json` file.
+The response of this request will contain an "x-recording-id" header. Save the value in `recording_id` variable.
+
+```sh
+scenarios=specifications/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/scenarios
+body='{ 
+  "x-recording-file": "'${scenarios}'/recording.json",
+  "x-recording-assets-file": "'${scenarios}'/assets.json"
+}'
+recording_id=$(curl -X POST -s -D - -d ${body} http://localhost:5000/Record/Start | grep 'x-recording-id' | awk '{print $2}' | sed 's/\r$//')
+```
+
+Now you can update the custom dictionary with the recording id and the test-proxy headers.
+
+```sh
+headers='{                                                 
+  "x-recording-upstream-base-uri": ["https://management.azure.com"],
+  "x-recording-id": ["'${recording_id}'"],
+  "x-recording-mode": ["record"]
+}'
+jq ".restler_custom_payload_header = $headers" dict.json > tmp.json && mv tmp.json dict.json 
+```
+
+### Recompile the configuration
+
+Now recompile the configuration to incorporate these changes.
+
+```sh
+dotnet $restler_bin/restler/Restler.dll compile config.json
+```
+
+## Run the tests
+
+Run the tests as describe in [QuickStart](./QuickStart.md#run-the-tests).
+
+## Stop the recording
+
+Once the tests have completed, stop the recording session by sending a `/Record/Stop` request to the test-proxy.
+```
+curl -X POST -D - -H  "x-recording-id: ${recording_id}" -H 'Content-Length: 0' http://localhost:5000/Record/Stop
+```
+
+Return to the terminal window where the test-proxy is running and press `Ctrl+C` to stop the test-proxy.
+
+The recording file can be found in the `.assets` folder.
+
+```sh
+find .assets -name recording.json
+```
+
+## Upload the Recording
+
+Check the recording file to ensure that all secrets have been removed.
+
```

Comment:
What should the user do if all secrets have not been removed?  Just remove them manually, or also notify someone (is this supposed to be done by the test proxy?)